registry/config.go

```diff
@@ -79,8 +79,11 @@ func init() {
 
 // IndexServerName returns the name of default index server.
 func IndexServerName() string {
-	if len(DefaultRegistries) < 1 {
-		return ""
+	// Set the default registry to docker.io unless there
+	// is one and only one registry entry in registries.conf.
+	// Note: docker.io is always added to the end of DefaultRegistries
+	if len(DefaultRegistries) != 2 {
```

Comment:
Should this be 1?

Reply:
Yikes, I really set this one aside for too long.  Just got back to this and spent some time testing.

To answer your question @rhatdan, no, this should be 2 as Docker always appends 'docker.io' to the list in memory.  So the code uses the registry from the table in registries.conf if and only if there is one and only one entry in that table in that  registries.conf file.  If there are two or more entries on the list in the table, then it goes to docker.io all the time.

After thinking about it further, I'm not sure that's right.  I think `docker login` should instead just always use the first entry in the list and ignore the rest.  Currently with this patch if there are more than two entries in the list it always defaults to docker.io even if the first entry in the list is valid such as 'quay.io'.

Given that the associated BZ has closed due to the fact that problematic default registry 'registry.access.redhat.com' has changed to a new v2 registry.  I think this PR should be closed and then the "Login Succeeded" code could be added to  registry/auth.go in a separate PR.  I believe that bit emulated Docker-ce's `docker login` functionality.